week1/project/src/components/ProductList/ProductList.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+
+const ProductList = ({ products, selectedCategory, onProductClick }) => {
+  const filteredProducts = Array.isArray(products)
+    ? selectedCategory
+      ? products.filter((product) => product.category === selectedCategory)
+      : products
+    : [];
```

Comment:
🤩

Reply:
A small tip (something you do not have to change now): you can also destructure the `category`:

```suggestion
  const filteredProducts = Array.isArray(products)
    ? selectedCategory
      ? products.filter(({ category ) => category === selectedCategory)
      : products
    : [];
```

Reply:
Thank you for pointing out this improvement to make the code more concise. I will apply it. I wasn't aware that I could directly request the property I am interested in for each product, instead of passing each object with all its properties.